File: frontend/src/types/log.ts
export type LogLevel = "DEBUG" | "INFO" | "WARNING" | "ERROR";

export const LOG_LEVELS: LogLevel[] = ["DEBUG", "INFO", "WARNING", "ERROR"];

export type LogSearchParams = {
  q: string;
  level: LogLevel | "";
};

export type Log = {
  level: LogLevel;
  message: string;
  service: string;
  timestamp: string;
  id_opensearch: string;
  index_opensearch: string;
};

export const LOG_PAGE_SIZE = 20;

export type LogLevelCounts = Record<LogLevel, number>;

export type LogSearchResult = {
  logs: Log[];
  total: number;
  page: number;
  page_size: number;
  level_counts: LogLevelCounts;
};

export const EMPTY_LEVEL_COUNTS: LogLevelCounts = {
  DEBUG: 0,
  INFO: 0,
  WARNING: 0,
  ERROR: 0,
};

File: backend/app/models/Log.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class LogLevel(str, Enum):
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"


class LogCreate(BaseModel):
    level: LogLevel
    service: str = Field(min_length=1, max_length=64)
    message: str = Field(min_length=1, max_length=2048)
    timestamp: datetime


class Log(LogCreate):
    id_opensearch: str
    index_opensearch: str = ""


class LogSearchResult(BaseModel):
    logs: list[Log]
    total: int
    page: int
    page_size: int
    level_counts: dict[str, int]
